tests/compiler/nonnullable.ts

```diff
@@ -0,0 +1,31 @@
+
+
+
+assert(nameof<NonNullable<u32>>() == nameof<u32>());
+
+assert(nameof<NonNullable<string[] | null>>() == nameof<string[]>());
+
+function takesGeneric<T>(t: T): void {
+  if (!isNullable<T>() || t != null){
+    // @ts-ignore
+    if  (isDefined(t.call)){
+      t.call();
+    }
+  }
+}
+
+function callsGeneric<T>(t: T): void {
+  if (t != null) {
+    takesGeneric<NonNullable<T>>(t); // Would be nice to have a `notNull<T>` here
+  }
+}
+
+class Tee {
+  call(): void {
+    let x: i8 = 0;
+  }
+}
+
+let y: Tee | null = new Tee();
+
+callsGeneric(y);
```

Comment:
Hmm, this test could be mush simpler and more clearer I guess. Also it not cover this case:
```ts
function takesGeneric<T>(t: NonNullable<T>): void {}

let str: string | null = null;
takesGeneric<string>(str); // error
```

For expected error you could add std_error field in `nonnullable.json` (See other compiler's tests)

Reply:
This case is already covered in the `nullable.ts` test.  I reorganized the test to make it simpler and uses a runtime check in addition to being compilable, so I need the file to compile.

Reply:
It not covered. Otherwise CI was failed. Also note AS not yet handle null safety in branches so you test just unsound

Reply:
I think the line

```ts
takesGeneric<NonNullable<T>>(t);
```

should only work when written as

```ts
takesGeneric<NonNullable<T>>(t!);
```

Otherwise I'd be worried :)

Reply:
I added this to my test and added one more that shows how you can check not null and then use cast to remove nullable.